cpp/CMakeLists.txt

```diff
@@ -935,7 +942,7 @@ add_dependencies(cudf jitify_preprocess_run)
 target_link_libraries(
   cudf
   PUBLIC CCCL::CCCL rmm::rmm rmm::rmm_logger $<BUILD_LOCAL_INTERFACE:BS::thread_pool>
-         spdlog::spdlog_header_only
+         spdlog::spdlog_header_only cudf_logger
```

Comment:
Should we stop explicitly require `spdlog` from now?

Reply:
Yes! Good call, I forgot to remove all references to spdlog when rebasing my changes. I'll update, there are a few spots where we can remove it.

Reply:
Done.